src/actions/channel_categories.ts

```diff
@@ -128,7 +128,10 @@ export function addChannelToInitialCategory(channel: Channel, setOnServer = fals
         }
 
         // Add the new channel to the Channels category on the channel's team
-        const channelsCategory = getCategoryInTeamByType(state, channel.team_id, CategoryTypes.CHANNELS);
+        let channelsCategory: ChannelCategory | undefined;
+        if (!categories.some((category) => category.channel_ids.some((channelId) => channelId === channel.id))) {
```

Comment:
Instead of only sometimes setting `channelsCategory`, could you instead return early if this condition is true? That'll help make it clear that we're not doing anything with the channel because it's already in a category.